### PR TITLE
Improved discovery

### PR DESCRIPTION
**Pull request by MatusKysel**
So idea is that node can report different IP (external IP), that should be use for discovery. I implemented basic mappings that will keep track of node ID and it's real ip (this could change) and real IP with external IP, so we know what should we report to the rest of the network

### CODE REVIEW

File: libraries/config/src/config.cpp [hunk 1]
```diff
@@ -23,10 +23,19 @@ Json::Value getConfigData(Json::Value root, std::vector<string> const &path, boo
   return root;
 }
 
-std::string getConfigDataAsString(Json::Value const &root, std::vector<string> const &path) {
+std::string getConfigDataAsString(Json::Value const &root, std::vector<string> const &path, bool optional = false,
+                                  std::string value = {}) {
```
Comment (JakubFornadel) [inline libraries/config/src/config.cpp:27]: what is this value argument good for ?

Comment (MatusKysel) [reply libraries/config/src/config.cpp:27]: default value, check the uint version

**Review by JakubFornadel — APPROVED** [review]

**Review by mfrankovi — APPROVED** [review]

**Pull request merged**